app/app.go

```diff
@@ -0,0 +1,115 @@
+package app 
+
+//modeled after basecoinapp in cosmos/cosmos-sdk/examples
+import (
+	// TODO: Change to import from FourthState repo (currently not on there)
+	types "plasma-mvp-sidechain/types" //points to a local package
+
+	bam "github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	dbm "github.com/tendermint/tmlibs/db"
+	cmn "github.com/tendermint/tmlibs/common"
+	crypto "github.com/tendermint/go-crypto"
+	"github.com/tendermint/tmlibs/log"
+
+	"github.com/tendermint/go-amino" // Not necessary once switched to RLP
+	//"github.com/ethereum/go-ethereum/rlp" // TODO: Change from amino to RLP
+
+)
+
+const (
+	appName = "plasmaChildChain" // Can be changed
+)
+
+// Extended ABCI application
+type ChildChain struct {
+	*bam.BaseApp // Pointer to the Base App
+
+	cdc *amino.Codec // Delete once changed to RLP
+	// keys to access the substores
+	capKeyMainStore *sdk.KVStoreKey //capabilities key to access main store from multistore
+	//Not sure if this is needed
+	capKeyIBCStore *sdk.KVStoreKey //capabilities key to access IBC Store from multistore
+
+	// Manage addition and deletion of unspent utxo's 
+	utxoMapper types.UTXOMapper
+}
+
+func NewChildChain(logger log.Logger, db dbm.DB) *ChildChain {
+	var app = &ChildChain{
+		BaseApp:			bam.NewBaseApp(appName, logger, db),
+		cdc:				MakeCodec(),
+		capKeyMainStore:	sdk.NewKVStoreKey("main"),
+		capKeyIBCStore:  	sdk.NewKVStoreKey("ibc"),
+	}
+
+	// define the utxoMapper
+	app.utxoMapper = types.NewUTXOMapper(
+		app.capKeyMainStore, // target store
+		// MYNOTE: may need to change proto
+		&types.BaseUTXOHolder{}, // UTXOHolder is a struct that holds BaseUTXO's
+							 // BaseUTXO implemented UTXO interface
+	)
+
+	// TODO: add handlers/router
+	// UTXOKeeper to adjust spending and recieving of utxo's
+	UTXOKeeper := types.NewUTXOKeeper(app.utxoMapper)
+	app.Router().
+		AddRoute("txs", types.NewHandler(UTXOKeeper))
+
+	// initialize BaseApp
+	// set the BaseApp txDecoder to use txDecoder with RLP
+	app.SetTxDecoder(app.txDecoder)
+	
+	// TODO: set initChainer?
+	// TO-UNDERSTAND: Not sure what mounting does yet
+	app.MountStoresIAVL(app.capKeyMainStore)
+	
+	// TO-UNDERSTAND: What does ante handler do, do i need to make a new one
```

Comment:
BaseApp is passing a Context and Tx into `AnteHandler` in its `runTx` method, which is used in both `checkTx()` and `deliverTx()`. I think we need would need an ante handler to take care of at least Tx signature verification.